packages/story-editor/src/components/panels/document/taxonomies/FlatTermSelector.js

```diff
@@ -129,6 +157,37 @@ function FlatTermSelector({ taxonomy, canCreateTerms }) {
         <Tags.Description id={`${taxonomy.slug}-description`}>
           {taxonomy.labels.separate_items_with_commas}
         </Tags.Description>
+        {mostUsed.length > 0 && (
+          <WordCloud.Wrapper>
+            <WordCloud.Heading>
+              {__('Most Used', 'web-stories')}
+            </WordCloud.Heading>
+            <WordCloud.List>
+              {mostUsed.map((term, i) => (
+                <WordCloud.ListItem key={term.id}>
+                  <WordCloud.Word
+                    onClick={() => {
+                      if (terms[taxonomy.restBase]?.includes(term.id)) {
+                        return;
+                      }
+                      addTermToSelection(taxonomy, term);
+                    }}
```

Comment:
can you move this to use a callback?

Reply:
reference this earlier thread here:
https://github.com/google/web-stories-wp/pull/9211#discussion_r719463109

Let me know what you think after you read through it